main/src/library/MutList.flix

```diff
@@ -833,6 +871,8 @@ namespace MutList {
     ///
     /// Applies `f` to all the elements in `v`.
     ///
+    /// pub def foreachWithIndex(f: (Int32, a) -> Unit \ { ef - Write(r) }, v: MutList[a, r]): Unit \ { ef, Read(r) }
+    ///
     pub def foreachWithIndex(f: (Int32, a) -> Unit \ ef, v: MutList[a, r]): Unit \ { ef, Read(r) } =
         let MutList(ra, rl) = v;
         let a = deref ra;
```

Comment:
A MutMap has an adjust function:

```
pub def adjust!(f: v -> v \ ef, k: k, m: MutMap[k, v, r]): Unit \ { ef, Read(r), Write(r) } with Order[k] =
```

Question 1: What does the corresponding function look like for `MutList`?
Question 2: Is it interesting?
Question 3: What if we change `f` to be `v -> Option[v]` where `None` would be interpreted as dropping that index in the `MutList`?

Reply:
1. Would say something along the lines of:  
    ```scala
    def adjust!(f: a -> a \ ef, i: Int32, l: MutList[a, r]): Unit \ { ef, Read(r), Write(r) } = {
        let g: (Int32, a) -> a \ ef = (j,x) -> {
            if (j == i) {
                f(x)
            } else {
                x
            }
        };
        MutList.transformWithIndex!(g, l)
    }
    ```
    This however make me realize, `MutList.transformWithIndex!` expects pure function, which might be somewhat odd - what if we wanted to read from some region and replace the element at `i` with that?
2. This could perhaps be an example of a benign write to the MutList
3. This would instead make it dangerous, similar to pop!